install_config/configuring_authentication.adoc

```diff
@@ -55,9 +59,47 @@ configuration file.
 xref:../architecture/additional_concepts/authorization.adoc#roles[Roles] need
 to be assigned to administer the setup with an external user.
 ====
+ 
+[[identity-providers-ansible]]
+== Configuring Identity Providers with Ansible
 
-[[identity-providers]]
+For initial xref:../install_config/install/advanced_install.adoc#install-config-install-advanced-install[advanced
+installations], the
+xref:../install_config/configuring_authentication.adoc#DenyAllPasswordIdentityProvider[Deny All] identity provider is configured by default,
+though it can be
+xref:../install_config/install/advanced_install.adoc#configuring-ansible[overridden during installation] using the
+xref:../install_config/install/advanced_install.adoc#configuring-cluster-variables[`*openshift_master_identity_providers*` parameter], which is configurable in the inventory file.
+
+.Example Identity Provider Configuration with Ansible
+====
+
+----
+# htpasswd auth
+openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
+# Defining htpasswd users
+#openshift_master_htpasswd_users={'user1': '<pre-hashed password>', 'user2': '<pre-hashed password>'
+# or
+#openshift_master_htpasswd_file=<path to local pre-generated htpasswd file>
 
+# Allow all auth
+#openshift_master_identity_providers=[{'name': 'allow_all', 'login': 'true', 'challenge': 'true', 'kind': 'AllowAllPasswordIdentityProvider'}]
+
+# LDAP auth
+#openshift_master_identity_providers=[{'name': 'my_ldap_provider', 'challenge': 'true', 'login': 'true', 'kind': 'LDAPPasswordIdentityProvider', 'attributes': {'id': ['dn'], 'email': ['mail'], 'name': ['cn'], 'preferredUsername': ['uid']}, 'bindDN': '', 'bindPassword': '', 'ca': '', 'insecure': 'false', 'url': 'ldap://ldap.example.com:389/ou=users,dc=example,dc=com?uid'}]
+# Configuring the ldap ca certificate
+#openshift_master_ldap_ca=<ca text>
+# or
+#openshift_master_ldap_ca_file=<path to local ca file to use>
+
+# Available variables for configuring certificates for other identity providers:
```

Comment:
@detiber @abutcher @sdodson are we missing directives for github, gitlab, and google? Are these providers even configurable via ansible? If not we should make a note of this.